website/docs/constrainttemplates.md

```diff
@@ -0,0 +1,155 @@
+---
+id: constrainttemplates
+title: Constraint Templates
+---
+
+ConstraintTemplates define a way to validate some set of Kubernetes objects in Gatekeeper's Kubernetes [admission controller](https://kubernetes.io/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/).  They are made of two main elements:
+
+1. [Rego](https://www.openpolicyagent.org/docs/latest/#rego) code that defines a policy violation
+2. The schema of the accompanying `Constraint` object, which represents an instantiation of a `ConstraintTemplate`
+
+## `v1` Constraint Template
+
+In release version 3.6.0, Gatekeeper included the `v1` version of `ConstraintTemplate`.  Unlike past versions of `ConstraintTemplate`, `v1` requires the Constraint schema section to be [structural](https://kubernetes.io/blog/2019/06/20/crd-structural-schema/).
+
+Structural schemas have a variety of [requirements](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#specifying-a-structural-schema).  One such requirement is that of the `type` field to be defined for each level of the schema.
```

Comment:
"is that of the `type` field to be defined for each level of the schema." -> "is that the `type` field must be defined for each level of the schema"